src/vm/callingconvention.h

```diff
@@ -1701,6 +1701,17 @@ inline BOOL HasRetBuffArg(MetaSig * pSig)
     return argit.HasRetBuffArg();
 }
 
+#ifdef UNIX_X86_ABI
+inline BOOL HasRetBuffArgUnmanagedFixup(MetaSig * pSig)
+{
+    WRAPPER_NO_CONTRACT;
+    CorElementType type = pSig->GetReturnTypeNormalized();
+
+    // For UNIX_X86_ABI and unmanaged function, we always need RetBuf if the return type is VALUETYPE
+    return type == ELEMENT_TYPE_VALUETYPE;
```

Comment:
This will return false for a simple struct like `struct X { int x; }` because of it will get normalized to int, but you want it to return true.

I think you need something like:

`pSig->GetReturnTypeHandleThrowing().GetVerifierCorElementType() == ELEMENT_TYPE_VALUETYPE`